### PR TITLE
Add lifecycle hook and updated samples for firebaserules_release

### PR DESCRIPTION
**Pull request by rainshen49**
Update samples for Firebase Security Rules to address https://github.com/hashicorp/terraform-provider-google/issues/13235

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

**Comment by shuyama1** [comment]
Sorry, I didn't have to time to review this last Friday. Reassigning a new reviewer as I'll be out for ~2 weeks. Thanks @ScottSuarez!

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: output.log
[the remark this answers is not in the document]
Comment (rainshen49) [reply output.log:1]: Oops. Done.

**Comment by rainshen49** [comment]
Ping @ScottSuarez this is ready for review!

**Comment by ScottSuarez** [comment]
Could join the GoogleCloudOrganization so our CI can detect you? Join [GoogleCloudPlatform org](https://github.com/GoogleCloudPlatform) through go/github/orgs

[github.com/orgs/GoogleCloudPlatform/people?query=rainshen49](https://github.com/orgs/GoogleCloudPlatform/people?query=rainshen49)

**Comment by ScottSuarez** [comment]
Looks like in the generated diff the samples are getting completely deleted.

https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7779-old..auto-pr-7779

I'm guessing this is because you relocated the samples to the release subdirectory. Wondering what the intention is there?

**Comment by rainshen49** [comment]
🤔 That wasn't intentional. Is it better now?

**Comment by ScottSuarez** [comment]
Looks like they weren't running before and are now thanks to 

https://github.com/GoogleCloudPlatform/magic-modules/pull/7779/commits/687743aaaafb331c7071b0003c74abd9fe1a99d3

Running them is the right move.. The error is because our version of terraform doesn't support this hook.
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/containers/gcb-terraform-vcr-tester/Dockerfile#L15-L16

`replace_triggered_by` was introduced in `1.2.

**Comment by rainshen49** [comment]
Why are tests in 1.1.8 when the latest version is 1.4.6? I also found that the registry documentation says we work with Terraform 0.13+ https://registry.terraform.io/providers/hashicorp/google-beta/latest 🤦 , so I'm absolutely clueless here.

**Comment by ScottSuarez** [comment]
> Why are tests in 1.1.8 when the latest version is 1.4.6? I also found that the registry documentation says we work with Terraform 0.13+ [registry.terraform.io/providers/hashicorp/google-beta/latest](https://registry.terraform.io/providers/hashicorp/google-beta/latest) 🤦 , so I'm absolutely clueless here.

Upgrading the terraform version in our repository is a significant effort. Tests can fail due to changes in terraform core. Could you omit usage of `replace_triggered_by` ? Is there an alternative behavior recommended?

**Comment by rainshen49** [comment]
tl;dr: we've tried many things in https://github.com/hashicorp/terraform-provider-google/issues/13235, and `replace_triggered_by` was the only solution that doesn't require any manual steps or breaking change in the provider.

The alternative is to make ruleset_name an immutable field, thus forcing a replacement, but this is worse because
1. It's a breaking change
2. It isn't how the underlying API works. The underlying API allows updates, and some users still want that behavior.

**Comment by rainshen49** [comment]
Done. I'd like to get rid of this tech debt when the version gets upgraded since the tests run are not exactly the same as the docs. Is there a way to add this item to the list of things to track for an upcoming upgrade?

On a separate note, what's the established schedule for updating the Terraform binary in tests? 1.1.8 feels quite far from 1.4.6. If the tests break as a result of the upgrade, doesn't that mean users out there are viewing outdated/non-functional samples?

**Comment by ScottSuarez** [comment]
After talking with our team, could you revert to your previous state of change. I'm going to update our internal version to 1.2.5

**Comment by ScottSuarez** [comment]
https://github.com/GoogleCloudPlatform/magic-modules/pull/7899

**Comment by ScottSuarez** [comment]
okay ! checked in. Could you force-push us back to`687743aaaafb331c7071b0003c74abd9fe1a99d3` ?

**Comment by rainshen49** [comment]
OK, done. Thanks for updating!

**Review by ScottSuarez — APPROVED** [review]
